applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -108,6 +120,79 @@ static struct bt_audio_discover_params audio_source_discover_param[CONFIG_BT_MAX
 static int discover_source(struct bt_conn *conn);
 #endif /* CONFIG_STREAM_BIDIRECTIONAL */
 
+/**
+ * @brief  Get the common presentation delay for all headsets.
+ *
+ * @param  mode         The mode of search for the presentation delay
+ * @param  index        The index of the headset to test against
+ * @param  pres_dly_us  A pointer where the presentation delay is stored
+ *
+ * @return 0 if successful, -EINVAL on an error
+ */
+static int headset_pres_delay_find(enum le_audio_pres_search mode, uint8_t index,
+				   uint32_t *pres_dly_us)
+{
+	uint32_t pres_dly_min = headsets[index].sink_ep->qos_pref.pd_min;
+	uint32_t pres_dly_max = headsets[index].sink_ep->qos_pref.pd_max;
+	uint32_t pref_dly_min = headsets[index].sink_ep->qos_pref.pref_pd_min;
+	uint32_t pref_dly_max = headsets[index].sink_ep->qos_pref.pref_pd_max;
+
+	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
+		if (headsets[i].sink_ep != NULL) {
+			pres_dly_min = MAX(pres_dly_min, headsets[i].sink_ep->qos_pref.pd_min);
+			pres_dly_max = MIN(pres_dly_max, headsets[i].sink_ep->qos_pref.pd_max);
+			pref_dly_min = MAX(pref_dly_min, headsets[i].sink_ep->qos_pref.pref_pd_min);
+			pref_dly_max = MIN(pref_dly_max, headsets[i].sink_ep->qos_pref.pref_pd_max);
+		}
+	}
+
+	switch (mode) {
+	case AUDIO_PRES_DELAY_MODE_MIN:
+		*pres_dly_us = pres_dly_min;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_MAX:
+		*pres_dly_us = pres_dly_max;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_PREF_MIN:
+		*pres_dly_us = pref_dly_min;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_PREF_MAX:
+		*pres_dly_us = pref_dly_max;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_LOCAL:
+		if (IN_RANGE(CONFIG_BT_AUDIO_LOCAL_PRESENTATION_DELAY_US, pres_dly_min,
+			     pres_dly_max)) {
+			*pres_dly_us = CONFIG_BT_AUDIO_LOCAL_PRESENTATION_DELAY_US;
+		} else {
+			LOG_WRN("Preferred local presentaion delay outside of range");
+
+			if (pres_dly_max < CONFIG_BT_AUDIO_LOCAL_PRESENTATION_DELAY_US) {
+				*pres_dly_us = pres_dly_max;
+
+				LOG_WRN("Selecting maximum common delay: %d us ", pres_dly_max);
```

Comment:
Is it maybe best to prioritize selecting minimum common?

Reply:
This is not prioritizing one over the other, it's simply taking the closest. If the "local" is outside the range it will either be larger or smaller than the range.